### PR TITLE
PREAPPS-6863: Classic UI - Display Files shared with me

### PR DESCRIPTION
**Pull request by miteshsavani810**
- Added Files shared with me functionality. 
- 'Files shared with me' folder appears only when sharing and document editor is enabled. 
- Prevent user to rename/delete 'Files shared with me' folder and it's properties by disabling context menu on this folder. 
- Showing 'Download' option for this folder's item. 
- 'Open' and 'Download' options get disable when multiple items are selected.
- Preventing user to move other folder item to this folder using drag and drop.

### CODE REVIEW

File: WebRoot/js/zimbraMail/share/model/ZmFolder.js [hunk 1]
```diff
@@ -140,6 +141,7 @@ ZmFolder.QUERY_NAME[ZmFolder.ID_AUTO_ADDED]		= "Emailed Contacts";
 ZmFolder.QUERY_NAME[ZmOrganizer.ID_BRIEFCASE]	= "briefcase";
 ZmFolder.QUERY_NAME[ZmFolder.ID_CHATS]			= "chats";
 ZmFolder.QUERY_NAME[ZmFolder.ID_SYNC_FAILURES]	= "Error Reports";
+ZmFolder.QUERY_NAME[ZmFolder.ID_FILE_SHARED_WITH_ME]	= "Files shared with me";
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/share/model/ZmFolder.js:144]: please check if folder name gets translated or not

Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/model/ZmFolder.js:145]: Folder name translation is handling by https://github.com/Zimbra/zm-web-client/pull/709/files#diff-063af5f64bad5434cc8f4edf1ee6b098b1984d0a3049fdad361dab0de0015c7dR115 this code.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 2]
```diff
@@ -400,6 +418,47 @@ function(parent, num) {
...
+		ZmOperation.DELETE ];
+
+	for (var i=0; i< fileOperations.length; i++) {
+		op = parent.getOp(fileOperations[i]);
+		if (op) {
+			op.setVisible(show);
+		}
+	}
+
+	if(parent &&  parent instanceof ZmActionMenu){
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Done

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 3]
```diff
@@ -400,6 +418,47 @@ function(parent, num) {
...
+		for (var k=0; k< parent.opList.length; k++) {
+			if (parent.opList[k] == "SEP") {
+				parent._children._array[k].setVisible(show);
+			}
+		}
+	}
+
+	if (parent && parent instanceof ZmButtonToolBar) {
+		parent.getActionsButton().setVisible(show);
+
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Removed.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 4]
```diff
@@ -1009,7 +1079,7 @@ function(items){
         item = AjxUtil.isArray(items) ? items[0] : items;
         restUrl = item.getRestUrl();
         restUrl += ( restUrl.indexOf('?') == -1 ) ? "?" : "&";
-        restUrl += "disp=a"+(item.version ? "&ver="+item.version : "");
+        restUrl += "disp=a"+(item.version && item.folderId != ZmFolder.ID_FILE_SHARED_WITH_ME ? "&ver="+item.version : "");
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/model/ZmBriefcaseItem.js [hunk 5]
```diff
@@ -205,6 +205,33 @@ function() {
 	return briefcase && briefcase.link;
 };
 
+ZmBriefcaseBaseItem.prototype.getDocServerUrl =
+function(callback){
+
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/model/ZmBriefcaseItem.js]: Removed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 6]
```diff
@@ -144,6 +144,7 @@ function(parent) {
         }
         headers.push(
                 new DwtListHeaderItem({field:ZmItem.F_LOCK, icon: "Padlock", width:ZmDetailListView.COLWIDTH_ICON, name:ZmMsg.lock}),
+				new DwtListHeaderItem({field:ZmItem.F_SHARES, icon: "SharedContact", width:ZmDetailListView.COLWIDTH_ICON, name:ZmMsg.lock}),
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 7]
```diff
@@ -314,7 +317,10 @@ function(item, colIdx) {
     html[idx++] = "</table>";
 
     html[idx++] = "<table width=100% class='BottomRow'><tr>";
-    html[idx++] = "<td style='vertical-align:middle;padding-left:50px;'>";
+	html[idx++] = "<td style='padding-left:28px;' width=20 id='" + this._getFieldId(item, ZmItem.F_SHARES)+"'> ";
+    idx =   this._getImageHtml(html, idx, (item.shares && item.shares.length > 0 ? "SharedContact" : "Blank_16") , this._getFieldId(item, ZmItem.F_SHARES));
+	html[idx++] = "</td>";
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 8]
```diff
@@ -400,6 +418,47 @@ function(parent, num) {
...
+		ZmOperation.EDIT_FILE,
+		ZmOperation.CHECKIN,
+		ZmOperation.CHECKOUT,
+		ZmOperation.DISCARD_CHECKOUT,
+		ZmOperation.RESTORE_VERSION,
+		ZmOperation.DETACH_WIN,
+		ZmOperation.SEP,
+		ZmOperation.DELETE ];
+
+	for (var i=0; i< fileOperations.length; i++) {
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Done.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 9]
```diff
@@ -271,13 +272,30 @@ function(parent, num) {
...
     var isAdmin = briefcase && briefcase.isAdmin(); 
 
+	if (isFileSharedWithMeFolder) {
+		// file shared with me folder selected.
+		if(parent &&  parent instanceof ZmActionMenu){
+			parent.getOp(ZmOperation.OPEN_FILE) && parent.getOp(ZmOperation.OPEN_FILE).setVisible(isItemSelected && !isMultiFolder);
+		}
+		parent.getOp(ZmOperation.OPEN_FILE).setVisible(true);
+		parent.enable(ZmOperation.OPEN_FILE, num == 1 && appCtxt.get(ZmSetting.DOCUMENT_EDITOR_ENABLED));
+		parent.enable(ZmOperation.SAVE_FILE, num == 1 && (!isFolderSelected || isBriefcaseItemSelected));
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Actually, "Files shared with me" folder will never have 'folder' and 'revision' item in the list. We are not allowing user to share 'revision' file. So I removed the checking of `isFolderSelected`, `isBriefcaseSelected` and just checking the number of items are selected.

[the remark this answers is not in the document]
Comment (k-kato) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: I got it. Thanks.

**Review by k-kato — APPROVED** [review]
LGTM

File: WebRoot/js/zimbraMail/briefcase/model/ZmBriefcase.js [hunk 10]
```diff
@@ -83,6 +83,7 @@ function() {
 ZmBriefcase.prototype.getIcon = 
 function() {
 	if (this.nId == ZmOrganizer.ID_ROOT)	{ return null; }
+	if (this.nId == ZmOrganizer.ID_FILE_SHARED_WITH_ME)	{ return "SharedContact"; }
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/model/ZmBriefcase.js:86]: SharedContact ???

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 11]
```diff
@@ -400,6 +409,47 @@ function(parent, num) {
     }
 };
 
+// Show / hide file toolbar and context menu options.
+ZmBriefcaseController.prototype.setFileOperations =
+function(parent, show) {
+	var fileOperations = [
+		ZmOperation.OPEN_FILE,
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js:416]: move this to zimlet

File: WebRoot/js/zimbraMail/briefcase/model/ZmBriefcase.js (hunk 10 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/model/ZmBriefcase.js:86]: Expected shareIcon is already present in the repo with "SharedContact" name.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js (hunk 11 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Done. Moved OPEN_FILE operation to zimlet code.

**Comment by silentsakky** [comment]
Need to fix conflicts

**Comment by miteshsavani810** [comment]
> Need to fix conflicts

Fixed.

**Review by silentsakky — APPROVED** [review]

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 12]
```diff
@@ -271,13 +272,30 @@ function(parent, num) {
...
 	var isReadOnly = briefcase ? briefcase.isReadOnly() : false;
 	var isMultiFolder = (noOfFolders > 1);
 	var isItemSelected = (num>0);
 	var isZimbraAccount = appCtxt.getActiveAccount().isZimbraAccount;
 	var isMailEnabled = appCtxt.get(ZmSetting.MAIL_ENABLED);
     var isAdmin = briefcase && briefcase.isAdmin(); 
 
+	if (isFileSharedWithMeFolder) {
+		// file shared with me folder selected.
+		if(parent &&  parent instanceof ZmActionMenu){
```
[the remark this answers is not in the document]
Comment (k-kato) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js:286]: It has been reverted. Please check it again.

**Review by k-kato — APPROVED** [review]
LGTM.
My comment is a minor thing.

**Comment by miteshsavani810** [comment]
> LGTM. My comment is a minor thing.

Fixed.

**Review by silentsakky — APPROVED** [review]

**Pull request merged**